### PR TITLE
[Merged by Bors] - chore(SetTheory/Ordinal/Basic): golf `Ordinal.lift` lemmas

### PR DESCRIPTION
**Pull request by vihdzp**
We define `Ordinal.liftInitialSeg` earlier and use it to more easily prove the other theorems on `lift`.

---

- [x] depends on: #17481

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
Looks sensible, thanks!

maintainer merge

File: Mathlib/SetTheory/Ordinal/Basic.lean [hunk 1]
```diff
@@ -657,59 +681,25 @@ theorem lift_one : lift 1 = 1 :=
...
-          lift_mk_eq.{u, max u v, max u v}] at e'
-        cases' e' with f
-        have g := RelIso.preimage f s
-        haveI := (g : f ⁻¹'o s ↪r s).isWellOrder
-        have := lift_type_eq.{u, max u v, max u v}.2 ⟨g⟩
-        rw [lift_id, lift_umax.{u, v}] at this
-        exact ⟨_, this⟩)
-    e
+theorem mem_range_lift_of_le {a : Ordinal.{u}} {b : Ordinal.{max u v}} (h : b ≤ lift.{v} a) :
+    b ∈ Set.range lift.{v} :=
```
Comment (fpvandoorn) [inline Mathlib/SetTheory/Ordinal/Basic.lean:688]: This looks like an `iff`?

File: Mathlib/SetTheory/Ordinal/Basic.lean [hunk 2]
```diff
@@ -1420,6 +1410,16 @@ theorem card_eq_zero {o} : card o = 0 ↔ o = 0 := by
 theorem card_eq_one {o} : card o = 1 ↔ o = 1 := by
   simpa using card_eq_nat (n := 1)
 
+theorem mem_range_lift_of_card_le {a : Cardinal.{u}} {b : Ordinal.{max u v}}
+    (h : card b ≤ Cardinal.lift.{v, u} a) : b ∈ Set.range lift.{v, u} := by
```
Comment (fpvandoorn) [inline Mathlib/SetTheory/Ordinal/Basic.lean:1414]: This looks like an `iff`?

**Review by fpvandoorn — APPROVED** [review]
Otherwise LGTM

**Comment by fpvandoorn** [comment]
Nevermind, I didn't read clearly.
Do we have `range lift = Iio univ` or something?

bors merge

**Pull request closed without merging**